### PR TITLE
Fix concurrent token renewal issues

### PR DESCRIPTION
**Pull request by AlexTugarev**
## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/10762

## How to test
1. Use git with the gitlab.com integration for a while, no issues with tokens should occur.

## Release Notes

```release-note
Token renewals should not run concurrently.
```

## Werft options:

- [x] /werft with-preview

### CODE REVIEW

File: components/server/src/user/token-service.ts [hunk 1]
```diff
@@ -52,9 +52,9 @@ export class TokenService implements TokenProvider {
                 `No token found for user ${identity.authProviderId}/${identity.authId}/${identity.authName}!`,
             );
         }
-        const refreshTime = new Date();
-        refreshTime.setTime(refreshTime.getTime() + 30 * 60 * 1000);
-        if (token.expiryDate && token.expiryDate < refreshTime.toISOString()) {
+        const aboutToExpireTime = new Date();
+        aboutToExpireTime.setTime(aboutToExpireTime.getTime() + 5 * 60 * 1000);
+        if (token.expiryDate && token.expiryDate < aboutToExpireTime.toISOString()) {
```
Comment (andrew-farries) [inline components/server/src/user/token-service.ts:57]: Why was this renamed?

**Review by andrew-farries — APPROVED** [review]
How does this work between multiple `server` replicas? This promise cache is per-server - does that matter?

/hold

File: components/server/src/user/token-service.ts (hunk 1 above)
Comment (AlexTugarev) [reply components/server/src/user/token-service.ts:57]: Renaming was not necessary, but that reflect the purpose of this time, while `refreshTime` does not. 

Also note the change from 30 to 5 minutes before expiration.
